packages/salesforcedx-utils-vscode/test/unit/helpers/utils.test.ts

```diff
@@ -7,20 +6,99 @@
 
 describe('getConfigSource', () => {
-
   it('should extract a JSON string from larger string and then return as an object', async () => {
-
-    const exampleJsonString = JSON.stringify({ name: 'exampleName', error: 'exampleError' });
+    const exampleJsonString = JSON.stringify({
+      name: 'exampleName',
+      error: 'exampleError'
+    });
     const exampleString = `junk text <junk +text junk text ${exampleJsonString} junk text junk text junk text`;
 
     const testParse = extractJsonObject(exampleString);
 
     expect(testParse.name).to.equal('exampleName');
     expect(testParse.error).to.equal('exampleError');
+  });
+});
+
+describe('flushFilePath', () => {
+  it('should call fs.realpathSync.native() to resolve a path', async () => {
+    const filePath = 'c:\\Users\\temp\\exampleFile.js';
+    const realpathSyncNativeStub = sinon
+      .stub(fs.realpathSync, 'native')
+      .returns(filePath);
+
+    const result = flushFilePath(filePath);
+
+    expect(realpathSyncNativeStub.called).to.equal(true);
+    expect(realpathSyncNativeStub.calledOnce).to.equal(true);
+    expect(realpathSyncNativeStub.args[0][0]).to.equal(filePath);
+
+    realpathSyncNativeStub.restore();
+  });
+
+  it('should return a path when a path is passed in', async () => {
+    const filePath = 'c:\\Users\\temp\\exampleFile.js';
+    const realpathSyncNativeStub = sinon
+      .stub(fs.realpathSync, 'native')
+      .returns(filePath);
+
+    const result = flushFilePath(filePath);
+
+    expect(result).to.equal(filePath);
+
+    realpathSyncNativeStub.restore();
+  });
+
+  it('should return an empty string when an empty sting is passed in', async () => {
+    const filePath = 'c:\\Users\\temp\\exampleFile.js';
+    const realpathSyncNativeStub = sinon
+      .stub(fs.realpathSync, 'native')
+      .returns('');
+
+    const result = flushFilePath(filePath);
+
+    expect(result).to.equal('');
+
+    realpathSyncNativeStub.restore();
+  });
+
+  it('should validate the correct path is returned when running on Windows', async () => {
+    const filePath = 'c:\\Users\\User Name\\foo.cls';
+    const realpathSyncFilePath = 'C:\\Users\\User Name\\foo.cls';
+    const realpathSyncNativeStub = sinon
+      .stub(fs.realpathSync, 'native')
+      .returns(realpathSyncFilePath);
+    const processPlatformStub = sinon.stub(process, 'platform').value('win32');
 
+    const result = flushFilePath(filePath);
+
+    expect(result).to.equal(filePath);
+
+    processPlatformStub.restore();
+    realpathSyncNativeStub.restore();
   });
+});
+
+describe('flushFilePaths', () => {
+  it('should return the same paths that are passed in', async () => {
+    const filePaths = ['file.js', 'file.js', 'file.js'];
+    const realpathSyncNativeStub = sinon
+      .stub(fs.realpathSync, 'native')
+      .returns(filePaths[0]);
 
+    const result = flushFilePaths(filePaths);
+
+    expect(result).to.equal(filePaths);
+
+    realpathSyncNativeStub.restore();
```

Comment:
Would this restore() be better placed in an afterEach() block?  Or do you prefer to keep it here inside each test case block?

Reply:
For the moment, there's only one test, so I think keeping it central to the test is better.  If there were more tests, and that stub was used in eery test (and the stubbed value never changed), then placing it in `beforeEach` would be a better place, but for here & now, probably best to keep it where it is.

Reply:
+1
Seems you could move the stub into a beforeEach and the restore into an afterEach FTW

Reply:
hum... I can't tell from the formatting but the `flushFilePath` and `flushFilePaths` describe blocks both appear to have tests that create and restore the following stub
```const realpathSyncNativeStub = sinon
      .stub(fs.realpathSync, 'native')
      .returns(realpathSyncFilePath);
...
      realpathSyncNativeStub.restore();
```
should those not be consolidated? I typically lean towards following patterns in tests to make them easier to grok for both new and existing developers so I'd prefer we put the stub/restore in before/after each blocks. This could be at the top level describe if that makes sense or just in a describe wrapper for all `flushFilePath*` describe blocks.

Reply:
oh..above I mean I couldn't tell if both describe blocks were wrapped in a single top level describe.

Reply:
@gbockus-sf maybe we should take this off line & discuss f2f.  realpathSyncNativeStub can't be hoisted or shared, b/c the mocked return value changes from function to function.

Reply:
@jeffb-sfdc Sure I'll ping ya in slack

Reply:
I wrapped everything in a `describe()`, added a `beforeEach()` and `afterEach()`, and moved realpathSyncNativeStub